martor/tests/tests.py

```diff
@@ -1,9 +1,23 @@
-from django.test import TestCase, override_settings
+import sys
+from importlib import reload
+
 from django.contrib.auth.models import User
+from django.core.signals import setting_changed
+from django.test import TestCase, override_settings
+from django.urls import resolve, reverse, NoReverseMatch, clear_url_caches
+
 from martor.utils import markdownify
+from martor.views import markdownfy_view, markdown_imgur_uploader, markdown_search_user
 
 
 class SimpleTest(TestCase):
+    def _on_settings_changed(self, sender, **kwargs):
+        # Reload settings.py and urls.py when @override_settings is called
+        clear_url_caches()
+        reload(sys.modules["martor.settings"])
+        reload(sys.modules["martor.urls"])
+        reload(sys.modules["martor.tests.urls"])
```

Comment:
This is a bit tricky, here: `@override_settings` doesn't force reload of every declared `urlpatterns` so we need to manually force it in order to test the dynamic adding of paths according to `settings.py`.